### ISSUE TITLE
Unable to view thumbs of .jpg or .png in media browser or file tree

### ISSUE THREAD

**Issue by muzzwood** [modxcms/revolution#15502]
### Summary

Using the latest git version of 3.x just now, thumbs are not showing for .jpg or .png files in either the file tree or media browser.
.svg thumbs work.

I'm also getting errors in the log: 
```
[2021-03-06 15:52:18] (ERROR @ /var/www/html/core/src/Revolution/modPhpThumb.php : 312) PHP warning: Cannot modify header information - headers already sent by (output started at /var/www/html/core/src/Revolution/Sources/modMediaSource.php:1499)
```

![Screenshot from 2021-03-05 22-54-58](https://user-images.githubusercontent.com/5160368/110199731-ccbb0400-7e94-11eb-95b2-146095c706ae.png)

![Screenshot from 2021-03-06 15-56-18](https://user-images.githubusercontent.com/5160368/110199737-d3e21200-7e94-11eb-9f90-b8b230da21e1.png)

I only have the default filesystem media source. No others were created.

### Step to reproduce
Hover over an image in the file tree.

### Observed behavior
Thumbs are broken and error displays in log

### Expected behavior
Thumbs shouldn't be broken :P

### Environment
MODX 3.x latest git, nginx, php7.4, mysql8, Firefox

**Comment by Ibochkarev** [comment]
Possibly related: https://github.com/modxcms/revolution/pull/15481

**Comment by Ibochkarev** [comment]
@digitalpenguin Please do not use [3.x] in the title

**Comment by Mark-H** [comment]
Can you please check the output of that phpthumb URL? The warning in the log indicates something was already in the output, which is most likely a php notice or warning indicating another problem.

Looking at the referenced line (modMediaSource line 1499), I'm guessing it's a notice for `url doesn't exist in array` or something like that:

```
        $properties = $this->getPropertyList();
        if ($properties['url']) {
            $src = $properties['url'] . DIRECTORY_SEPARATOR . ltrim($src, DIRECTORY_SEPARATOR);
        }
```

... which begs the question if the url is supposed to be there or how to handle that if not.

**Comment by muzzwood** [comment]
`$properties` array at line 1499:

```
[2021-03-07 11:55:30] (ERROR @ /var/www/html/core/src/Revolution/Sources/modMediaSource.php : 1499) Array
(
    [basePath] => 
    [basePathRelative] => 1
    [baseUrl] => 
    [baseUrlRelative] => 1
    [allowedFileTypes] => 
    [imageExtensions] => jpg,jpeg,png,gif,svg
    [thumbnailType] => png
    [thumbnailQuality] => 90
    [visibility] => public
    [skipFiles] => .svn,.git,_notes,nbproject,.idea,.DS_Store
    [source] => 1
    [w] => 400
    [h] => 0
    [f] => png
    [q] => 90
    [t] => 1614955345
)

```

This is a log of the phpThumb `$debugMessages` array just before it does an `exit;` - (plus the warnings from just above)
```
[2021-03-07 13:37:42] (ERROR @ /var/www/html/core/src/Revolution/modPhpThumb.php : 304) PHP warning: Cannot modify header information - headers already sent by (output started at /var/www/html/core/src/Revolution/Sources/modMediaSource.php:1499)
[2021-03-07 13:37:42] (ERROR @ /var/www/html/core/src/Revolution/modPhpThumb.php : 312) PHP warning: Cannot modify header information - headers already sent by (output started at /var/www/html/core/src/Revolution/Sources/modMediaSource.php:1499)
[2021-03-07 13:37:42] (ERROR @ /var/www/html/core/src/Revolution/modPhpThumb.php : 322) Array
(
    [0] => phpThumb() v1.7.16-202012161640 in file "phpthumb.class.php" on line 234
    [1] => setCacheDirectory() starting with config_cache_directory = "/var/www/html/core/cache/phpthumb/" in file "phpthumb.class.php" on line 1048
    [2] => setting config_cache_directory to $this->realPathSafe(/var/www/html/core/cache/phpthumb) = "/var/www/html/core/cache/phpthumb" in file "phpthumb.class.php" on line 1076
    [3] => setSourceFilename(/var/www/html/assets/q5d9Fsz.png) set $this->sourceFilename to "/var/www/html/assets/q5d9Fsz.png" in file "phpthumb.class.php" on line 299
    [4] => $AvailableImageOutputFormats = array(text;ico;bmp;wbmp;gif;webp;png;jpeg) in file "phpthumb.class.php" on line 1012
    [5] => $this->thumbnailFormat set to $this->config_output_format "jpeg" in file "phpthumb.class.php" on line 1023
    [6] => $this->thumbnailFormat set to $this->f "png" in file "phpthumb.class.php" on line 1030
    [7] => $this->thumbnailQuality set to "90" in file "phpthumb.class.php" on line 1040
    [8] => SetCacheFilename() _src set from md5($this->sourceFilename) "/var/www/html/assets/q5d9Fsz.png" = "d8a507f9a39159905c6beec6adbf5358" in file "phpthumb.class.php" on line 3616
    [9] => SetCacheFilename() _par set from md5(_farC_w400_dpi150) in file "phpthumb.class.php" on line 3655
)
```

If you mean something differently to checking the output of the URL, where/how would I do that? Thanks!

**Comment by muzzwood** [comment]
Ah, so the thumbs are actually being generated in the cache. And the above output is from when it appears to be accessing the cache.

I deleted those thumb files in the cache and tried again. There's more to look at now:

```
[2021-03-07 13:54:14] (ERROR @ /var/www/html/core/src/Revolution/modPhpThumb.php : 304) PHP warning: Cannot modify header information - headers already sent by (output started at /var/www/html/core/src/Revolution/Sources/modMediaSource.php:1499)
[2021-03-07 13:54:14] (ERROR @ /var/www/html/core/src/Revolution/modPhpThumb.php : 312) PHP warning: Cannot modify header information - headers already sent by (output started at /var/www/html/core/src/Revolution/Sources/modMediaSource.php:1499)
[2021-03-07 13:54:14] (ERROR @ /var/www/html/core/src/Revolution/modPhpThumb.php : 322) Array
(
    [0] => phpThumb() v1.7.16-202012161640 in file "phpthumb.class.php" on line 234
    [1] => setCacheDirectory() starting with config_cache_directory = "/var/www/html/core/cache/phpthumb/" in file "phpthumb.class.php" on line 1048
    [2] => setting config_cache_directory to $this->realPathSafe(/var/www/html/core/cache/phpthumb) = "/var/www/html/core/cache/phpthumb" in file "phpthumb.class.php" on line 1076
    [3] => setSourceFilename(/var/www/html/assets/q5d9Fsz.png) set $this->sourceFilename to "/var/www/html/assets/q5d9Fsz.png" in file "phpthumb.class.php" on line 299
    [4] => $AvailableImageOutputFormats = array(text;ico;bmp;wbmp;gif;webp;png;jpeg) in file "phpthumb.class.php" on line 1012
    [5] => $this->thumbnailFormat set to $this->config_output_format "jpeg" in file "phpthumb.class.php" on line 1023
    [6] => $this->thumbnailFormat set to $this->f "png" in file "phpthumb.class.php" on line 1030
    [7] => $this->thumbnailQuality set to "90" in file "phpthumb.class.php" on line 1040
    [8] => SetCacheFilename() _src set from md5($this->sourceFilename) "/var/www/html/assets/q5d9Fsz.png" = "d8a507f9a39159905c6beec6adbf5358" in file "phpthumb.class.php" on line 3616
    [9] => SetCacheFilename() _par set from md5(_farC_w400_dpi150) in file "phpthumb.class.php" on line 3655
    [10] => $this->sourceFilename set to "/var/www/html/assets/q5d9Fsz.png" in file "phpthumb.class.php" on line 933
    [11] => $this->cache_filename already set, skipping SetCacheFilename() in file "phpthumb.class.php" on line 3572
    [12] => starting ExtractEXIFgetImageSize() in file "phpthumb.class.php" on line 3409
    [13] => getimagesize(/var/www/html/assets/q5d9Fsz.png) says image is 1920x1080 in file "phpthumb.class.php" on line 3419
    [14] => $this->useRawIMoutput=true after checking $UnAllowedParameters in file "phpthumb.class.php" on line 1655
    [15] => ImageMagickThumbnailToGD() aborting because ImageMagickCommandlineBase() failed in file "phpthumb.class.php" on line 2386
    [16] => ImageMagickThumbnailToGD() failed in file "phpthumb.class.php" on line 3458
    [17] => SetOrientationDependantWidthHeight() starting with "1920"x"1080" in file "phpthumb.class.php" on line 3388
    [18] => SetOrientationDependantWidthHeight() setting w="400", h="0" in file "phpthumb.class.php" on line 3404
    [19] => exif_thumbnail() does not exist, cannot extract EXIF thumbnail in file "phpthumb.class.php" on line 3513
    [20] => EXIF thumbnail extraction: (size=0; type=""; 0x0) in file "phpthumb.class.php" on line 3517
    [21] => starting SourceImageToGD() in file "phpthumb.class.php" on line 3781
    [22] => starting ImageCreateFromFilename(/var/www/html/assets/q5d9Fsz.png) in file "phpthumb.class.php" on line 3696
    [23] => ImageCreateFromFilename found ($getimagesizeinfo[2]==3) in file "phpthumb.class.php" on line 3705
    [24] => Calling imagecreatefrompng(/var/www/html/assets/q5d9Fsz.png) in file "phpthumb.class.php" on line 3713
    [25] => Not using EXIF thumbnail data because $this->gdimg_source is already set in file "phpthumb.class.php" on line 3849
    [26] => CalculateThumbnailDimensions() starting with [W,H,sx,sy,sw,sh] initially set to [1920,1080,,,,] in file "phpthumb.class.php" on line 3260
    [27] => CalculateThumbnailDimensions() starting with [x,y,w,h] initially set to [0,0,1920,1080] in file "phpthumb.class.php" on line 3275
    [28] => CreateGDoutput() returning canvas "400x225" in file "phpthumb.class.php" on line 3383
    [29] => ImageResizeFunction($o, $s, 0, 0, 0, 0, 400, 225, 1920, 1080) in file "phpthumb.class.php" on line 4473
    [30] => memory_get_usage() after copy-resize = 10529680 in file "phpthumb.class.php" on line 433
    [31] => memory_get_usage() after imagedestroy = 1661968 in file "phpthumb.class.php" on line 435
    [32] => AntiOffsiteLinking() says this is allowed in file "phpthumb.class.php" on line 2578
    [33] => skipping AlphaChannelFlatten() because ($this->thumbnailFormat == "png") in file "phpthumb.class.php" on line 2642
    [34] => GenerateThumbnail() completed successfully in file "phpthumb.class.php" on line 447
    [35] => CleanUpCacheDirectory() set to purge (30.0 days; 100.00 MB; 10,000 files) in file "phpthumb.class.php" on line 751
    [36] => CleanUpCacheDirectory() skipped because $CacheDirOldFilesSize is empty (phpthumb_functions::GetAllFilesInSubfolders(/var/www/html/core/cache/phpthumb) found no files) in file "phpthumb.class.php" on line 784
    [37] => imageinterlace($this->gdimg_output, 1) in file "phpthumb.class.php" on line 483
    [38] => RenderOutput() attempting imagepng($this->gdimg_output) in file "phpthumb.class.php" on line 486
    [39] => Specified thumbnailQuality "90" is outside the accepted range (0-9, or -1). Using 6 as default value. in file "phpthumb.class.php" on line 530
    [40] => RenderOutput() completing with $this->outputImageData = 73323 bytes in file "phpthumb.class.php" on line 600
    [41] => RenderToFile(/var/www/html/core/cache/phpthumb/modx3git.local__srcd8a507f9a39159905c6beec6adbf5358_par5ec9be3d082e600179ae51f34f149e74_dat1614955345.png) succeeded in file "phpthumb.class.php" on line 629
)


```

**Comment by alroniks** [comment]
@Ibochkarev Please, validate that the merged pull request fixed this issue.

**Comment by Ibochkarev** [comment]
Fixed in #15526 

https://user-images.githubusercontent.com/2138260/111520132-00920580-8782-11eb-9826-5694df0e7c79.mov
